### PR TITLE
Core separation

### LINKED ISSUE #3133: Separate VS core to NPM package
offering the VueStorefront code as NPM package 

- is easier to install
- easier to update
- hide complexity
- easier the API

Includes adding CLI

### PR DESCRIPTION
**Pull request by yanko-ivanov**
### Related issues
Not applicable

closes #3133

### Short description and why it's useful
This is the first step towards actually separating the VSF core into its own npm package.

### Screenshots of visual changes before/after (if there are any)
None applicable

### Which environment this relates to

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature
- [x] Change is breaking and a proper upgrade guideline is yet to be settled

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: core/build/webpack.base.config.ts [hunk 1]
```diff
@@ -110,26 +121,45 @@ export default {
...
-      path.resolve(__dirname, themesRoot)
+      path.resolve(__dirname, themesRoot),
+      path.resolve(__dirname, moduleRoot)
     ],
     extensions: ['.js', '.vue', '.gql', '.graphqls', '.ts'],
     alias: {
       // Main aliases
       'config': path.resolve(__dirname, './config.json'),
-      'src': path.resolve(__dirname, '../../src'),
+      'src': path.resolve(__dirname, config.projectRoot + '/src'),
```
Comment (filrak) [inline core/build/webpack.base.config.ts]: @yanko-ivanov why not a single alias to `@vue-storefront/core' ?

Comment (yanko-ivanov) [reply core/build/webpack.base.config.ts]: Great question and one I've thought about much. For some reason, that didn't work in my experience. As you know, the @vue-storefront/core file path should be picked up from the package.json onwards, as it's in the node_modules. But for some reason deep linking doesn't work, and you have to explicitly resolve it. Prime examples are - as seen below - lib/ and lib/modules, data-resolver and data-resolver/types, and i18n as a whole. 
I know that this leaves us in a situation where the base webpack config is to be tracked and updated whenever the file structure changes, which isn't ideal, but I couldn't find a better solution.

File: sfui.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/* Here you can override global SCSS variables */
```
Comment (filrak) [inline sfui.scss]: can you remove storefront ui for now? It's not used anywhere

**Comment by pkarw** [comment]
I'm afraid we won't be able to finalize this PR and with the new CLI (1.12.1) it's no longer at a high priority bc. CLI solves the issues this separation was intended to solve

**Pull request closed without merging**